### PR TITLE
Support List-detail view

### PR DESCRIPTION
**Pull request by viduycheung**
## Description

Support List-detail view for better task info view experience.

## Screencast

CleanShot 2022-03-14 at 13 49 08@2x

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed in the `media` folder

### CODE REVIEW

**Comment by viduycheung** [comment]
Add new command <Search Tasks>
CleanShot 2022-03-16 at 10 36 15@2x

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @viduycheung 👋 

Thanks for the update and for making the extension better and better 💪 I tested every command and couldn't find anything not working.

I have some feedback for you to address:

- In taskItem.tsx, `OpenAction` is deprecated and [Action.Open](https://developers.raycast.com/api-reference/user-interface/actions#action.open) should be used instead

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Comment by viduycheung** [comment]
@pernielsentikaer  fix it 🕺

**Comment by viduycheung** [comment]
@pernielsentikaer Any update here 😿...

File: extensions/ticktick/src/components/searchFilter.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+interface Props {
+  onFilterChange: (filterId: string) => void;
+  filterSections: FilterSection[];
+}
+
+const SearchFilter: React.FC<Props> = (props) => {
+  const { filterSections, onFilterChange } = props;
+  return (
+    <List.Dropdown tooltip="Select Drink Type" storeValue={true} onChange={onFilterChange}>
```
Comment (pernielsentikaer) [inline extensions/ticktick/src/components/searchFilter.tsx:22]: Drink? 😄

File: extensions/ticktick/src/utils/sort.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+function _compareDigit(d1: number, d2: number) {
+  if (d1 === d2) {
+    return 0;
+  } else {
+    return d1 > d2 ? 1 : -1;
+  }
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function _defined(d1: any, d2: any) {
```
Comment (pernielsentikaer) [inline extensions/ticktick/src/utils/sort.ts:12]:
```suggestion
function _defined(d1: string, d2: string) {
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @viduycheung 👋 

I'm so sorry, I'm not sure how I lost track of this one, I must have removed the notifications, sorry again! 😄

In osScript.ts you're using the deprecated `ToastStyle`, could you correct that to match the new version of [Toast](https://developers.raycast.com/api-reference/feedback/toast)

In the same file, lines 68-86 and lines 104-122 are 100% the same, could be a function instead to ensure it's actually doing the same and make it easier to maintain.

You have a lot `eslint-disable` going on, is most of them necessary? 

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Comment by viduycheung** [comment]
@pernielsentikaer 
all issues have been fixed

I remove unnecessary `eslint-disable` as much as I can, some `no-explicit-any` I think is necessary because the data struct has come from apple script.

**Comment by pernielsentikaer** [comment]
Hi @viduycheung 👋 

Much netter, thanks! We have added support for [categories](https://developers.raycast.com/basics/prepare-an-extension-for-store#categories), [screenshots](https://developers.raycast.com/basics/prepare-an-extension-for-store#screenshots) and [changelog](https://developers.raycast.com/basics/prepare-an-extension-for-store#version-history) – do you have time for adding this too before merging?

**Comment by viduycheung** [comment]
@pernielsentikaer Done.

**Review by pernielsentikaer — APPROVED** [review]
Awesome @viduycheung 🔥 

I'm going to approve it, nice update 💪

**Comment by sxn** [comment]
hey @viduycheung - nice work!

One question: it feels a bit weird that `Search Tasks` doesn't show anything unless I type something. Is it not possible to fetch all tasks without user input? In that case, I'd suggest adding an [EmptyView](https://developers.raycast.com/api-reference/user-interface/list#list.emptyview) explaining this. 😺 

https://user-images.githubusercontent.com/1155589/161069526-5883f156-e881-4810-925d-8670fa9638f6.mov

File: extensions/ticktick/CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# TickTick Changelog
+
+## [Added] - 2022-3-31
```
Comment (sxn) [inline extensions/ticktick/CHANGELOG.md:3]:
```suggestion
## [Added] - 2022-03-31
```

**Comment by sxn** [comment]
hi @viduycheung - was just a fluke :D re-running the pipeline fixed it. 🎉

**Pull request merged**